app/src/org/odk/collect/android/logic/ArchivedFormRemoteRestore.java

```diff
@@ -0,0 +1,131 @@
+package org.odk.collect.android.logic;
+
+import android.widget.Toast;
+
+import org.commcare.android.database.user.models.User;
+import org.commcare.android.framework.CommCareActivity;
+import org.commcare.android.tasks.DataPullTask;
+import org.commcare.android.tasks.FormRecordCleanupTask;
+import org.commcare.android.tasks.templates.CommCareTaskConnector;
+import org.commcare.android.util.SessionUnavailableException;
+import org.commcare.dalvik.activities.FormRecordListActivity;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.commcare.util.CommCarePlatform;
+import org.javarosa.core.services.locale.Localization;
+
+/**
+ * Only used for developer debugging.
+ *
+ * Load saved form instances manually from a xml payload.
+ *
+ * @author Phillip Mates (pmates@dimagi.com).
+ */
+public class ArchivedFormRemoteRestore<A extends CommCareActivity> {
```

Comment:
Ah ok, makes sense on leaving it in it's own class then, but I think I'd advocate removing the generics. Significantly overcomplicating a class for a potential test that is likely not to get written doesn't seem like the right choice. Good to merge once that's changed, and possibly leaning out the test resource file I commented on? Sorry I didn't offer any useful comments on what to take out, but don't want to mislead you- Clayton could definitely help with that.